themes/default/content/docs/reference/service-api/_index.md

```diff
@@ -0,0 +1,1465 @@
+
+## Required Request Headers
+
+TODO: Add more info.
+
+```
+Accept: application/vnd.pulumi+3
+Content-Type: application/json
+```
+
+<!-- ###################################################################### -->
+
+## Stacks
+
+### List Stacks
+
+```
+GET https://api.pulumi.com/api/user/stacks
+```
+
+#### Parameters
+
+| Parameter | Type | In | Description |
+| --------- | ---------- | ---------- | ---------- |
+| `organization` | string | query | **Optional.** organization name to filter stacks by |
+| `project` | string | query | **Optional.** organization name to filter stacks by |
+| `tagName` | string | query | **Optional.** tag name to filter stacks by |
+| `tagValue` | string | query | **Optional.** tag value to filter stacks by |
+
+#### Example
+
+```bash
+curl \
+  -H 'Accept: application/vnd.pulumi+3' \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: token pul-abcdefghijklmnopqrstuvwxyz' \
+  https://api.pulumi.com/api/user/stacks?organization={organization}&tagName={tagName}&tagValue={tagValue}
+```
+
+#### Default response
+
+```
+Status: 200 OK
+```
+
+```
+{
+  "stacks": [
+    {
+      "orgName": "org1",
+      "projectName": "project1",
+      "stackName": "dev",
+      "lastUpdate": 1612481387,
+      "resourceCount": 0
+    },
+    {
+      "orgName": "org2",
+      "projectName": "project2",
+      "stackName": "prod",
+      "lastUpdate": 1605474464,
+      "resourceCount": 0
+    }
+  ]
+}
+```
+
+### Get Stack
+
+```
+GET https://api.pulumi.com/api/stacks/{organization}/{project}/{stack}
+```
+
+#### Parameters
+
+| Parameter | Type | In | Description |
+| --------- | ---------- | ---------- | ---------- |
+| `organization` | string | path | organization name |
+| `project` | string | path | project name |
+| `stack` | string | path | stack name |
+
+#### Example
+
+```bash
+curl \
+  -H 'Accept: application/vnd.pulumi+3' \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: token pul-abcdefghijklmnopqrstuvwxyz' \
+  https://api.pulumi.com/api/stacks/{organization}/{project}/{stack}
+```
+
+#### Default response
+
+```
+Status: 200 OK
+```
+
+```
+{
+  "orgName": "demo",
+  "projectName": "demo-aws-ts-webserver",
+  "stackName": "dev-user1",
+  "currentOperation": {
+    "kind": "update",
+    "author": "user1",
+    "started": 1615872156
+  },
+  "activeUpdate": "d333a711-4aa0-402f-be6d-72af9665fc37",
+  "tags": {
+    "gitHub:owner": "pulumi",
+    "gitHub:repo": "customer-engineering",
+    "pulumi:description": "A minimal AWS TypeScript Pulumi program",
+    "pulumi:project": "demo-aws-ts-webserver",
+    "pulumi:runtime": "nodejs",
+    "pulumi:secrets_provider": "service",
+    "vcs:kind": "github.com",
+    "vcs:owner": "pulumi",
+    "vcs:repo": "customer-engineering"
+  },
+  "version": 91
+}
+```
+
+### Get Stack State
```

Comment:
"Stack State" might not be the best way to phrase this. (Though I'm not 100% sure how we refer to this elsewhere in our documentation.)

This returns the "checkpoint file" of the Pulumi Stack. So "stack state" is perfectly clear and accurate, but just maybe not the most precise/technical term to use.

Reply:
If you ever do get questions about the specific schema/format, we want to make clear that it isn't meant to be a "human-friendly" representation of the resource state. And is instead, our own, internal view of the resources. Generally it's good enough for direct consumption, e.g. checking `output` properties and what not.

But there are no guarantees about what the `id` field means, e.g., as it is dependent on the implementation of the actual resource provider. So for AWS it is _usually_ the ARN, but with no guarantees. Also, when it comes to things like "stack references" or other oddities, they are flagged by having having a GUID as a property name which would be very confusing to people looking at the data...

Anyways, we just want to be a little vague that we are exposing the state information in a specific file format and not a general purpose/always-super-friendly way.

Reply:
Given we refer to this as the "state" in many other places in the docs, I feel that introducing the term "checkpoint" would be confusing.